src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -312,22 +276,28 @@ object WorkflowDescriptor {
                                       options: WorkflowOptions,
                                       backend: Backend,
                                       conf: Config): ErrorOr[WorkflowDescriptor] = {
-    val gcsInterface = GoogleCloudStorage.userAuthenticated(options) orElse GoogleCloudStorage.cromwellAuthenticated
-    val ioManager = backend match {
-      case _: JesBackend => gcsInterface match { // JesBackend only supports gcsInterface
-        case Success(x) => x
-        case Failure(t) => throw new Throwable("No GCS interface has been found. When running on JES Backend, Cromwell requires a google configuration to perform GCS operations.", t)
-      }
-      case _ => new IoManager(Seq(gcsInterface.toOption, Option(SharedFileSystemIoInterface.instance)).flatten)
-    }
+//    val gcsInterface = GoogleCloudStorage.userAuthenticated(options) orElse GoogleCloudStorage.cromwellAuthenticated
+    val gcsStorage = GoogleCloudStorage.userAuthenticated(options) orElse GoogleCloudStorage.cromwellAuthenticated
     val wfContext = new WorkflowContext(buildWorkflowRootPath(backend.rootPath(options), namespace.workflow.unqualifiedName, id))
-    val engineFunctions = backend.engineFunctions(ioManager, wfContext)
+    val gcsFileSystemProvider = gcsStorage map GcsFileSystemProvider.apply
+    val wfGcsFileSystem = gcsFileSystemProvider map { _.newFileSystem(wfContext.root) }
+    val fileSystems = backend match {
+      case _: JesBackend =>
+        val gcs = wfGcsFileSystem match {
+          case Success(x) => x
+          case Failure(t) => throw new Throwable("No GCS interface has been found. When running on JES Backend, Cromwell requires a google configuration to perform GCS operations.", t)
+        }
+        List(gcs, defaultFileSystem)
+      case _ => List(wfGcsFileSystem.toOption, Option(defaultFileSystem)).flatten
+    }
```

Comment:
this filesystem determination should really move to the backends soon, if not now